kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/jaxb/JaxbMarshaller.java

```diff
@@ -480,10 +486,19 @@ public String marshall(Object input) {
     public <T> T unmarshall(String input, Class<T> type) {
         try {
             Unmarshaller unmarshaller = getUnmarshaller();
+
+            //Disable XXE
+            SAXParserFactory spf = SAXParserFactory.newInstance();
+            spf.setNamespaceAware(true);
+            spf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
```

Comment:
@tkobayas Would it have sense to allow configuring this value? With default value set to true?
Thinking whether there may be some valid cases for using doctype declarations.

Reply:
@sutaakar  `disallow-doctype-decl` is considered as the safest way to prevent XXE (https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html#general-guidance).

A payload generated by kie-server-client API is the standard payload and it doesn't use DOCTYPE (also DOCTYPE is not used in drools documents). Even if there is a user who created a payload manually with DOCTYPE, I think we are right to say "we always disallow DOCTYPE in kie-server payload for security reasons". WDYT? @mariofusco @elguardian

Reply:
I honestly don't see any reason to send a payload with the DOCTYPE to the kie-server and, also considering the security risks it implies, I'd leave this as it is.

Reply:
ok, fine for me